litestar/_openapi/schema_generation/schema.py

```diff
@@ -350,533 +354,421 @@ def create_schema_for_annotation(annotation: Any) -> Schema | None:
     if isinstance(annotation, EnumMeta):
         return create_enum_schema(annotation)
 
-    return None
-
-
-def create_schema_for_optional_field(
-    field: SignatureField,
-    generate_examples: bool,
-    plugins: list[OpenAPISchemaPluginProtocol],
-    schemas: dict[str, Schema],
-    prefer_alias: bool,
-) -> Schema:
-    """Create a Schema for an optional SignatureField.
-
-    Args:
-        field: A signature field instance.
-        generate_examples: Whether to generate examples if none are given.
-        plugins: A list of plugins.
-        schemas: A mapping of namespaces to schemas - this mapping is used in the OA components section.
-        prefer_alias: Whether to prefer the alias name for the schema.
+    return Schema()
+
+
+@dataclass(frozen=True)
```

Comment:
Why dataclass? I think this class shouldn't be a dataclass. 

Is it computed once per app? 

Also, please add `slots`

Reply:
Mostly as a convenience to get `__init__` for free, plus the `frozen=True` doesn't hurt either. I tend to make most classes dataclasses these days for the free `__init__`, `__repr__`, `__eq__` etc, not just "dumb data" containers. Is there a downside?

I'll add `__slots__` (in 3.10+ that would be just `slots=True`).

Reply:
It's not a problem as is.